### PR TITLE
Register apps database migration

### PR DESCRIPTION
**Pull request by esoergel**
@dannyroberts
misc notes: https://docs.google.com/document/d/134Q2gfZUFBZOnrfQioo0m69fpfX509uYcGCWVPmq5_U/edit

### CODE REVIEW

File: corehq/couchapps/cloudcare_apps/views/view/map.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function(doc){
```
Comment (dimagimon) [inline corehq/couchapps/cloudcare_apps/views/view/map.js]: Missing name in function declaration.

File: corehq/couchapps/cloudcare_apps/views/view/map.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    if((doc.doc_type == 'Application') && doc.copy_of == null && doc.cloudcare_enabled) {
```
Comment (dimagimon) [inline corehq/couchapps/cloudcare_apps/views/view/map.js]: Use '===' to compare with 'null'.

File: corehq/couchapps/__init__.py [hunk 3]
```diff
@@ -7,8 +7,16 @@
...
-        settings.NEW_DOMAINS_DB),
+        settings.NEW_DOMAINS_DB, settings.NEW_APPS_DB),
+
+    # register these views to both the main db AND the apps db.
+    'app_translations_by_popularity': (None, settings.NEW_APPS_DB),
+    'exports_forms_by_app': (None, settings.NEW_APPS_DB),
+    'forms_by_app_info': (None, settings.NEW_APPS_DB),
+
+    # register just to the new apps db
+    'cloudcare_apps': settings.NEW_APPS_DB,
```
Comment (esoergel) [inline corehq/couchapps/__init__.py]: This is a little unorthodox - I'm copying an existing view here, then registering it only to the new db.  Then during the flip, we can switch `"cloudcare/cloudcare_apps"` to `"cloudcare_apps/view"`.  After the switch we can delete the cloudcare design doc (that's the only view).  This avoids reindexing the view in the main db, although we have other options.  Let's discuss tomorrow.

File: corehq/couchapps/cloudcare_apps/views/view/map.js (hunk 2 above)
Comment (esoergel) [inline corehq/couchapps/cloudcare_apps/views/view/map.js]: I think I can safely make that change without the two views functioning any different.

File: corehq/couchapps/cloudcare_apps/views/view/map.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+function(doc){
+    if((doc.doc_type === 'Application') && doc.copy_of == null && doc.cloudcare_enabled) {
```
Comment (dimagimon) [inline corehq/couchapps/cloudcare_apps/views/view/map.js]: Use '===' to compare with 'null'.

File: corehq/doctypemigrations/migrator_instances.py [hunk 5]
```diff
@@ -36,6 +36,23 @@
...
+    doc_types=(
+        'Application',
+        'ApplicationBase',
+        'CareplanConfig',
+        'DeleteApplicationRecord',
+        'DeleteFormRecord',
+        'DeleteModuleRecord',
+        'RemoteApp',
+        'SavedAppBuild',
+        'VersionedDoc',
```
Comment (esoergel) [inline corehq/doctypemigrations/migrator_instances.py]: Do I need to include "abstract" doc types like "ApplicationBase"?  It's not intended to be written directly, but you can totally do `ApplicationBase().save()`

File: settings.py [hunk 6]
```diff
@@ -1092,6 +1092,9 @@
 NEW_DOMAINS_DB = 'domains'
 DOMAINS_DB = NEW_DOMAINS_DB
 
+NEW_APPS_DB = 'apps'
+APPS_DB = None
```
Comment (benrudolph) [inline settings.py]: :eyes: omg i'm so excited for this

Comment (esoergel) [inline settings.py]: :snowman:

File: corehq/doctypemigrations/migrator_instances.py (hunk 5 above)
Comment (dannyroberts) [inline corehq/doctypemigrations/migrator_instances.py]: I'd say it's not strictly necessary, since there shouldn't be any docs with that doc type. If by some long-ago bug there were though, we'd want them to be moved to the new db, so I'd say it's fine to leave in there.

File: corehq/couchapps/__init__.py (hunk 3 above)
Comment (dannyroberts) [inline corehq/couchapps/__init__.py]: Heh, I mean if you really want to do this, you can always write

```python
'cloudcare_apps/view' if settings.NEW_APPS_DB == settings.APPS_DB else 'cloudcare/cloudcare_apps'
```

wherever `'cloudcare/cloudcare_apps'` is currently used.

Comment (dannyroberts) [inline corehq/couchapps/__init__.py]: However I think the plan I outlined here is simpler: https://github.com/dimagi/commcare-hq/pull/10022#issuecomment-174012914.

Rather than break `cloudcare/cloudcare_apps` out, just sync `_design/cloudcare` to apps db. (This is the only view in it, so it doesn't even need to be in the main db, just the apps db.) Once the migration's complete, we can then merge that other PR that merges this view with `app_manager/applications_brief`, and maximal optimality has been obtained

File: corehq/doctypemigrations/README.md [hunk 7]
```diff
@@ -228,3 +249,8 @@ Keep in mind that this will likely incur some reindexing overhead in the source
 
 If you run `$ ./manage.py run_doctype_migration user_db_migration --stats`, you
 should see that only the target database has these doc types in it now.
+
+
+### Stop syncing irrelevant views to the old db
+If you synced any `couchapps` views to both dbs for the migration, you should
+register those only to the new db (in `corehq/couchapps/__init__.py`).
```
Comment (dannyroberts) [inline corehq/doctypemigrations/README.md]: I'd maybe reword this. My convention was to sync to both `APPS_DB` and `NEW_APPS_DB`—before the migration they're different, after the migration they're the same. Thus no cleanup step is necessary except when down the road you remove the reference to NEW_APPS_DB entirely. (We've never done this step so far because I think it would screw up the blocking mirgrations, etc., and it's easy to keep around for a while before it becomes a pain.)

File: corehq/couchapps/__init__.py (hunk 3 above)
Comment (esoergel) [inline corehq/couchapps/__init__.py]: Ha, I'd forgotten about that guy (Mentally filed it away under general view cleanup, not *Application* view cleanup).  Yeah, I like that plan better - feels safer.

**Comment by esoergel** [comment]
Woo, tests passed!  @dannyroberts can you take a pass at the new changes I made?

**Comment by snopoke** [comment]
:fireworks:
